### DISCUSSION TITLE
crkbd rev4.0 with miryoku halves not working

### DISCUSSION THREAD

**Discussion by mdjohns** [manna-harbour/miryoku#372] [Q&A] [answered]
Hey all,

Attempting to get Miryoku set up on a new crkbd rev 4.0 and running into some trouble. I've flashed both halves and each half is correct when connected via USB. However, when connecting both via TRRS, the right half does not respond. Anything I could have missed?

Here are the steps I've taken:
* Fork `qmk_firmware`
* Pull in Miryoku as directed in https://github.com/manna-harbour/miryoku/discussions/287 (https://github.com/mdjohns/qmk_firmware/commit/4020c9f17d70d92532c486517042d20a26731c30)
  * This included resolving [the breaking change to key overrides](https://docs.qmk.fm/ChangeLog/20240825#key-override-keymap-c-signature-change-24120) as mentioned in https://github.com/manna-harbour/miryoku/discussions/287#discussioncomment-11423425
* Pull in the Miryoku 3x6_3 community layout https://github.com/mdjohns/qmk_firmware/commit/03935a910e7bdae8ccaf3f1b54b4a1527690317c 
* Build and flash as normal via `make crkbd/rev4_0/standard:manna-harbour_miryoku:flash MIRYOKU_NAV=VI`

**Comment by mdjohns** [comment, marked as answer]
Sharing the solution for posterity:
tl;dr: If you have a Corne v4, make sure that you're flashing the correct version.  

As expected this had nothing to do with Miryoku itself.
I purchased an assembled Corne v4 from AliExpress, which came pre-flashed with Via.
As mentioned above, I flashed Miryoku QMK and both halves worked separately, but not together.
I re-flashed the board with the firmware directly from [Foostan's firmware docs](https://github.com/foostan/crkbd/blob/main/docs/firmware/rev4/firmware_en.md#flash-the-firmware) and experienced the same issue.
I removed one side of the keyboard from the case to examine the version on the PCB and sure enough it was a 4.1.0.

After flashing the correct keyboard version things are working as expected.
